src/biogeochem/test/CNFireFactory_test/test_CNFireFactory.pf

```diff
@@ -0,0 +1,225 @@
+      expected_msg = "ABORTED: Unknown option for namelist item fire_method: UNSET"
+      @assertExceptionRaised(expected_msg)
+
+    end subroutine fire_method_not_set_fails
+
+    !-----------------------------------------------------------------------
+
+    @Test
+    subroutine fire_method_bad_fails(this)
+      class(TestCNFireFactory), intent(inout) :: this
+      character(100) :: expected_msg
+
+      call this%FireFactInit( fire_method = "ZZTOP")   ! Set to an invalid option
+      expected_msg = "ABORTED: Unknown option for namelist item fire_method: ZZTOP"
+      @assertExceptionRaised(expected_msg)
+
+    end subroutine fire_method_bad_fails
+
+    !-----------------------------------------------------------------------
+
+    @Test
+    subroutine nofire_with_fire_emis_fails(this)
+      class(TestCNFireFactory), intent(inout) :: this
+      character(100) :: expected_msg
+
+      call this%turn_fire_emis_on()
+      call this%FireFactInit( fire_method = "nofire")
+      expected_msg = "ABORTED: Having fire emissions on requires fire_method to be something besides nofire"
+      @assertExceptionRaised(expected_msg)
+
+    end subroutine nofire_with_fire_emis_fails
+
+    !-----------------------------------------------------------------------
+
+    @Test
+    subroutine spcase_with_fire_emis_fails(this)
+      use SatellitePhenologyMod, only : SatellitePhenologyInit
+      class(TestCNFireFactory), intent(inout) :: this
+      character(100) :: expected_msg
+
+      use_cn = .false.
+      call this%turn_fire_emis_on()
+      call SatellitePhenologyInit( bounds )
+      expected_msg = "ABORTED: Fire emission requires BGC to be on rather than a Satelitte Pheonology (SP) case"
+      @assertExceptionRaised(expected_msg)
+
+    end subroutine spcase_with_fire_emis_fails
+
+    !-----------------------------------------------------------------------
+
+    @Test
+    subroutine li2014_works(this)
+      class(TestCNFireFactory), intent(inout) :: this
+
+      call this%FireFactInit( fire_method = "li2014qianfrc")
+
+    end subroutine li2014_works
+
+    !-----------------------------------------------------------------------
+
+    @Test
+    subroutine li2016_works(this)
+      class(TestCNFireFactory), intent(inout) :: this
+
+      call this%FireFactInit( fire_method = "li2016crufrc")
+
+    end subroutine li2016_works
+
+    !-----------------------------------------------------------------------
+
+    @Test
+    subroutine li2021_works(this)
+      class(TestCNFireFactory), intent(inout) :: this
+
+      call this%FireFactInit( fire_method = "li2021gswpfrc")
+
+    end subroutine li2021_works
+
+    !-----------------------------------------------------------------------
+
+    @Test
+    subroutine li2024_works(this)
+      class(TestCNFireFactory), intent(inout) :: this
+
+      call this%FireFactInit( fire_method = "li2024gswpfrc")
+
+    end subroutine li2024_works
+
```

Comment:
We think we should remove the loop over all tests, and go for the simpler standalone tests. They will only be added one at a time.

Reply:
Also add a test for: li2024crujra

Reply:
Also needs a test for fire_emis on and that it also works.

Reply:
Or I could add a comment about the difference between the two, as an example of two ways to do it. I think I'll check with some others about best practice here.

Reply:
"Repetition of the individual tests above, but turning fire-emissions on and making sure it still works."